### ISSUE TITLE
Onboarding upgrade branch sync (Part 1)

### ISSUE THREAD

**Issue by fbwoolf** [blockstack/blockstack-app#2099]
This issue will track getting the changes we made on `feat/onboarding-upgrade` in sync with the current state of `dev`. Part 1 addresses changes made up to the new account home page.

Link to designs: https://www.figma.com/file/VupGh90FJtT0dcBj2Sh7bb/%F0%9F%93%8FSpec?node-id=3%3A8

**Comment by timstackblock** [comment]
@fbwoolf Anything testable for me on this issue?

**Comment by fbwoolf** [comment]
> @fbwoolf Anything testable for me on this issue?

You can definitely test that all the onboarding routes work as expected and reference the new designs to make sure I caught all the changes. I was waiting for the wallet release and then was going to merge it into dev ...if you want to wait and test it there?

**Comment by timstackblock** [comment]
Ok sounds good thanks for the heads up.

**Comment by fbwoolf** [comment]
As a side note, I did not change the `Allow Diagnostics` step to be a popup bc I wasn't sure it was supposed to be changed? @eugeniadigon do you know?

**Comment by ginny-d** [comment]
> As a side note, I did not change the `Allow Diagnostics` step to be a popup bc I wasn't sure it was supposed to be changed? @eugeniadigon do you know?

Is this the 'Help us improve' modal? Unfortunately though I just added illustrations on this part, I'm not aware of what was the context on why that was designed that way

**Comment by fbwoolf** [comment]
O, I think maybe I know why, @beguene I think it was moved to be the first step so the user has to approve it first?

**Comment by fbwoolf** [comment]
@beguene can you let me know on this tomorrow in your AM? I know you implemented this originally, why was it changed to full page and not a modal? Can/should I change it to match the new designs or was there a reason we moved away from it?

**Comment by beguene** [comment]
@fbwoolf Yes it's at the first step because we want the user approval first so we can then track everything we want to track for the onboarding. This was changed to a full page during my PR review since approval of analytics is a step now that must happen before the full onboarding. I think you can just keep the analytics approval first and then continue as per the design.

**Comment by fbwoolf** [comment]
@beguene thx for the feedback. That is what I thought must have happened. So, I'll leave it as the first step in full page view. @timstackblock ignore the modal design for the allow diagnostic step in the new design.

**Comment by fbwoolf** [comment]
This was merged into dev branch and can be tested with that build once it is ready.

**Comment by Eshwari007** [comment]
_I am following the flow in Figma under new account home page._
@fbwoolf On the New account design in Figma , notice that 


- Under Create stacks account

```
Already have a stacks account
Sign in with Secret Key or connect your Ledger
```
<img width="703" alt="Screen Shot 2022-01-27 at 4 52 32 PM" src="https://user-images.githubusercontent.com/38872890/151450223-25116993-bc15-4a61-a847-5b60b3bcb7bf.png">



However in the PR -
It listed as `I already have an account`
<img width="394" alt="Screen Shot 2022-01-27 at 4 52 15 PM" src="https://user-images.githubusercontent.com/38872890/151450310-92bc8b83-b1bc-4949-abe3-182bb52e68c6.png">

Wasnt sure if this was intentional?

**Comment by Eshwari007** [comment]
I was not able to view '`Help us improve` 'screen
<img width="312" alt="Screen Shot 2022-01-27 at 4 57 29 PM" src="https://user-images.githubusercontent.com/38872890/151450550-ee947704-38b3-4ae3-ba95-918449b3a4e5.png">

**Comment by Eshwari007** [comment]
<img width="664" alt="Screen Shot 2022-01-27 at 5 04 16 PM" src="https://user-images.githubusercontent.com/38872890/151451405-ac278def-da3e-4f95-babf-678c0b5a2927.png">


On this page:
The logo above _`_Your secret key_`_  looks different than the one on Figma
Also noticed that at the bottom of the page after '`I've backed it up`' the ' or Back it up later' missing from both popup wallet view and full window view

**Comment by Eshwari007** [comment]
The logo/picture on top of `set a password is missing
<img width="1096" alt="Screen Shot 2022-01-27 at 5 17 26 PM" src="https://user-images.githubusercontent.com/38872890/151453039-a09b7737-cb85-4c98-b8e8-0cb23f54f578.png">
<img width="759" alt="Screen Shot 2022-01-27 at 5 17 34 PM" src="https://user-images.githubusercontent.com/38872890/151453040-97669f17-3397-4207-b324-59513be8bfa0.png">

The initial onboarding screen  may need recentering too

**Comment by fbwoolf** [comment]
@Eshwari007 these update are separate issues in the repo. There is a PR being reviewed to update the welcome page already. I noted on that PR to make these updates there.

See PR here:
https://github.com/hirosystems/stacks-wallet-web/pull/2171#issuecomment-1022774232

**Comment by Eshwari007** [comment]
Should the `password strength is weak` appear when the user inputs a weak password or only appears upon selecting Done?
I do not see the the weak password while inputting the password 
<img width="286" alt="Screen Shot 2022-01-27 at 5 29 49 PM" src="https://user-images.githubusercontent.com/38872890/151454249-4441ac41-46e0-416b-8eab-2c529ff491e3.png">
 but instead only upon inputting weak password and selecting done 

<img width="648" alt="Screen Shot 2022-01-27 at 5 30 47 PM" src="https://user-images.githubusercontent.com/38872890/151454354-23f87487-4101-41d8-827a-0e7fd84854c6.png">

**Comment by fbwoolf** [comment]
> Should the `password strength is weak` appear when the user inputs a weak password or only appears upon selecting Done? I do not see the the weak password while inputting the password <img alt="Screen Shot 2022-01-27 at 5 29 49 PM" width="286" src="https://user-images.githubusercontent.com/38872890/151454249-4441ac41-46e0-416b-8eab-2c529ff491e3.png"> but instead only upon inputting weak password and selecting done
> 

What makes you think it should happen on input? In production it has always been on submit?

**Comment by Eshwari007** [comment]
> @Eshwari007 these update are separate issues in the repo. There is a PR being reviewed to update the welcome page already. I noted on that PR to make these updates there.
> 
> See PR here: [#2171 (comment)](https://github.com/hirosystems/stacks-wallet-web/pull/2171#issuecomment-1022774232)





@fbwoolf I will retest the issues above once PR # 2171 is merged. Believe Part 2 of onboarding will have Onboarding First,Gaia Hub, and App sign inflow as shown in Figma if my understanding is correct?

**Comment by Eshwari007** [comment]
> > Should the `password strength is weak` appear when the user inputs a weak password or only appears upon selecting Done? I do not see the the weak password while inputting the password <img alt="Screen Shot 2022-01-27 at 5 29 49 PM" width="286" src="https://user-images.githubusercontent.com/38872890/151454249-4441ac41-46e0-416b-8eab-2c529ff491e3.png"> 
> 
> What makes you think it should happen on input? In production it has always been on submit?

The way how the design is depicted without having any password entered in the second input box made me think if we have changed the logic as only password weak is shown on the first input box(as if a user entering in the first input box and the warning message appears). Also the way how `Password strength: Weak` label differs from design to implementation raised the question on my mind.
Hence I wanted to clarify the logic behind it.

**Comment by fbwoolf** [comment]
> 
> The way how the design is depicted without having any password entered in the second input box made me think if we have changed the logic as only password weak is shown on the first input box(as if a user entering in the first input box and the warning message appears). Also the way how `Password strength: Weak` label differs from design to implementation raised the question on my mind. Hence I wanted to clarify the logic behind it.
>
No, it will appear on submit as far as I know. Tbh, this design is just incomplete so I'm going to leave the error label as-is until the new designer can take a look at it throughout the app. This error label is reused quite often. There is no indication here about what the error label should look like on the `confirm` input so I just left them the same for now.

**Comment by fbwoolf** [comment]
Sorry @Eshwari007 for the confusion on this ...I was a bit confused myself when I went back to the designs and all the illustrations had changed. I wonder if we can have versions in Figma so that issues stay linked to the right designs? @andresgalante any thoughts on this?

**Comment by Eshwari007** [comment]
> Sorry @Eshwari007 for the confusion on this ...I was a bit confused myself when I went back to the designs and all the illustrations had changed. I wonder if we can have versions in Figma so that issues stay linked to the right designs? @andresgalante any thoughts on this?

Not a a prob @fbwoolf .You always give me a heads up on changes hence I was curious abt the design difference between what has been implemented vs design.

**Comment by fbwoolf** [comment]
@Eshwari007 I linked the onboarding updates PR to this issue. Feel free to close this in favor of all the design update issues linked to that PR if makes more sense to you. That branch should now match Figma designs linked in this issue.

**Comment by Eshwari007** [comment]
Thank you @fbwoolf ,will do the needful.
